evm/src/fuzz/strategies/param.rs

```diff
@@ -25,22 +25,24 @@ pub fn fuzz_param(param: &ParamType) -> impl Strategy<Value = Token> {
         // rejection sampling if it's determined the bias is too severe. Rejection sampling may
         // slow down tests as it resamples bad values, so may want to benchmark the performance
         // hit and weigh that against the current bias before implementing
-        ParamType::Int(n) => match n / 8 {
-            32 => any::<[u8; 32]>()
-                .prop_map(move |x| I256::from_raw(U256::from(&x)).into_token())
-                .boxed(),
-            y @ 1..=31 => any::<[u8; 32]>()
+        ParamType::Int(n) => {
+            // use uint strat but convert to int
+            let nc = *n;
+            super::UintStrategy::new(*n, vec![])
                 .prop_map(move |x| {
-                    // Generate a uintN in the correct range, then shift it to the range of intN
-                    // by subtracting 2^(N-1)
-                    let uint = U256::from(&x) % U256::from(2).pow(U256::from(y * 8));
-                    let max_int_plus1 = U256::from(2).pow(U256::from(y * 8 - 1));
-                    let num = I256::from_raw(uint.overflowing_sub(max_int_plus1).0);
-                    num.into_token()
+                    // TODO: Check the bounds here. We may miss 0 or a max/min value
+                    // if this is wrong I believe?
+                    if x > U256::from(2u32).pow((nc as u32 / 2).into()) {
```

Comment:
Shouldn't that be `nc - 1` rather than `nc / 2`

Reply:
Additionally I think it would work properly since the ValueTree for Unit simplifies the value by `value /= 2`. That means that if value is negative, on the 2nd simplification the value would go from negative to positive zone and the simplification will stop most likely.

I think it's better to implement a separate int strategy